lib/src/mocks/mock_http.dart

```diff
@@ -34,7 +34,9 @@ class MockHttp {
     MockHttpInternal._verifyRequestIsMock(request);
     final MockBaseRequest mockRequest = request;
     mockRequest.complete(response: response);
-    MockHttpInternal._pending.remove(request);
+    mockRequest.done.then((_) {
+      MockHttpInternal._pending.remove(request);
```

Comment:
This was an existing timing bug - it was trying to remove the pending request from the list before it was ever being added, since it doesn't get added until the request actually gets "sent"